util/set/set.go

```diff
@@ -7,9 +7,38 @@ package set
 // Set is a set of T.
 type Set[T comparable] map[T]struct{}
 
+// New returns a new Set.
```

Comment:
New generally returns a pointer in Go.

I'm not sure how much utility this constructor provides. The type is a map and we're not hiding that behind some opaque struct. Might as well reduce the number of was call sites can look.

Reply:
Sounds good; removed.